### PR TITLE
Render more config fields (#960)

### PR DESCRIPTION
**Pull request by beckjake**
Implement #960. Almost all fields in project/profile configs are now rendered.

Unrendered fields are 'on-run-end'/'on-run-start' in the root of a project config, and 'post-hook'/'pre-hook' under 'models' and 'seeds' in a project config.

Profile fields are rendered in a two-stage process (target, then targeted profile), to allow users to require nonexistent environment/cli variables on non-loaded targets without error.

The big config-side change, beyond the obvious rendering stuff, is having to pass the cli variables around to the various classmethod constructors so they can build a renderer.

### CODE REVIEW

File: dbt/config.py [hunk 1]
```diff
@@ -90,27 +90,94 @@ def colorize_output(config):
...
+        # model hooks
+        if first in {'seeds', 'models'}:
+            if 'pre-hook' in keypath or 'post-hook' in keypath:
+                return True
+
+        return False
+
+    @staticmethod
+    def _is_port_path(keypath):
+        return len(keypath) == 2 and keypath[-1] == 'port'
```
Comment (cmcarthur) [inline dbt/config.py]: this isn't used

File: dbt/config.py [hunk 2]
```diff
@@ -90,27 +90,94 @@ def colorize_output(config):
...
+
+    @staticmethod
+    def _convert_port(value, keypath):
+
+        if len(keypath) != 4:
+            return value
+
+        if keypath[-1] == 'port' and keypath[1] == 'outputs':
```
Comment (cmcarthur) [inline dbt/config.py]: i assume you meant to use `_is_port_path` here.

this is ok for now. thought: this should be on the adapter as part of "Adapters as Plugins"

File: dbt/config.py [hunk 3]
```diff
@@ -90,27 +90,94 @@ def colorize_output(config):
...
+                return True
+
+        return False
+
+    @staticmethod
+    def _is_port_path(keypath):
+        return len(keypath) == 2 and keypath[-1] == 'port'
+
+    @staticmethod
+    def _convert_port(value, keypath):
```
Comment (cmcarthur) [inline dbt/config.py]: oh... it looks like this also isn't used?

File: dbt/config.py [hunk 4]
```diff
@@ -90,27 +90,94 @@ def colorize_output(config):
...
+        # keypath is ignored.
+        # if it wasn't read as a string, ignore it
+        if not isinstance(value, compat.basestring):
+            return value
+
+        return dbt.clients.jinja.get_rendered(value, self.context)
+
+    def _render_profile_data(self, value, keypath):
+        result = self.render_value(value)
+        if len(keypath) == 1 and keypath[-1] == 'port':
```
Comment (cmcarthur) [inline dbt/config.py]: same comment from `_convert_port`: someday profile-specific rendering info should go on the adapter

Comment (beckjake) [reply dbt/config.py]: Adapters will have to provide their own credentials descriptions when they're plugins so I imagine I'll be forced to deal with this then.

**Pull request merged**